### ISSUE TITLE
Instance Segmentation training on a custom dataset (some questions)

### ISSUE THREAD

**Issue by Robotatron** [IDEA-Research/MaskDINO#28]
First of all, thanks for creating and making this model public!

I am training this model on a custom dataset (600 images with 7 classes) with my custom polygon annotations to train an Instance Segmentation model. Test evaluation is done on the same training dataset to make sure everything works.
Some questions I hope you could answer:

### 1. Learning rate (LR) on a single GPU
> We use total batch size 16 for all our models. If train on 1 GPU, you need to figure out learning rate and batch size by yourself

The default LR from the config is 0.0001 for 8x GPUs from your training scenario. Does that mean for a single GPU I have to divide the LR by 8? 

### 2. Some Instance Segmentation COCOEval mAP scores are suspiciously round numbers (e.g. 90.000 mAP)
Does it ever happen that mAP scores are such round float values down to the precision of a thousands after the dot? (e.g. 50.000 mAP )? It seems to me it highly unlikely to have a MAP score of x.000 since we are calculating IoU of a mask. 
Does that mean there is an issue with COCOeval metric calculation?
![image](https://user-images.githubusercontent.com/95919873/209755739-9eed9e64-2f98-467c-88d4-7f9786f5565c.png)

Now I even got 2x classes having mAP of 100. I do testing on the same dataset I am training on, but is it even realitic to have mAP of 100.000 for instance segmentation?

![image](https://user-images.githubusercontent.com/95919873/209756321-2ffde7bf-93ca-45e0-b9d9-ea04e5ded0f5.png)

### 3. RuntimeWarning: invalid value encountered in intersection
During training I get a lot of shapely warnings:
```/opt/conda/envs/detectron/lib/python3.9/site-packages/shapely/set_operations.py:133: RuntimeWarning: invalid value encountered in intersection return lib.intersection(a, b, **kwargs)```

Could it mean there is some issue with my polygon annotations? If so, is there an easy way to debug it? I manually double checked that all my polygon annotations are valid polygons (not self intersecting, closed polygon, etc). Is there something else I can do? This issue could be related to question 2).

Thanks in advance!

**Comment by FengLi-ust** [comment]
Hi,
1. You need to figure out the LR with the batch size you use, not the number of GPUs. We did not try using another batch size, maybe you can divide the LR by 8 when using batch size 2.
2. We did not observe such round numbers when trained on COCO, so I can not provide suggestions on this problem.
3. We did not encounter this problem before. Maybe you can print these wrong annotations or draw them on the images to see if they are wrong.

**Comment by FengLi-ust** [comment]
Issue closed. You can reopen this issue if you have further problems.
